dali/dfuXRefLib/XRefNodeManager.cpp

```
@@ -176,143 +145,182 @@ StringBuffer & CXRefNode::getLastModified(StringBuffer & str)
-        {
-            foundBranch = m_XRefTree->addPropTree("Found",createPTree());
-            commit();
-        }
-        StringBuffer tmpbuf;
-        m_found.setown(new CXRefFilesNode(*foundBranch,getName(tmpbuf).str(),rootDir));
-    }
-    return m_found.getLink();
+    return getOrCreateFilesNode<CXRefFilesNode>("Found", m_found);
 }
 
 IXRefFilesNode* CXRefNode::getOrphanFiles()
 {
-    if(!m_orphans.get())
-    {
-        IPropertyTree* orphanBranch = m_XRefTree->queryPropTree("Orphans");
-        if(orphanBranch == 0)
-        {
-            orphanBranch = m_XRefTree->addPropTree("Orphans",createPTree());
-            commit();
-        }
-        StringBuffer tmpbuf;
-        m_orphans.setown(new CXRefOrphanFilesNode(*orphanBranch,getName(tmpbuf).str(),rootDir));
-    }
-    return m_orphans.getLink();
+    return getOrCreateFilesNode<CXRefOrphanFilesNode>("Orphans", m_orphans);
 }
 
-StringBuffer &CXRefNode::serializeMessages(StringBuffer &buf)
+StringBuffer &CXRefNode::serializeBranch(const char *branchName, Owned<IPropertyTree> &cached, StringBuffer &buf)
 {
-    if(!m_messages.get())
+    buf.clear();
+    const char *xrefPath = m_XRefTree->queryProp("@xrefPath");
+
+    if (xrefPath && *xrefPath)
     {
-        IPropertyTree* messagesBranch = m_XRefTree->queryPropTree("Messages");
-        if(messagesBranch == 0)
+        // File-based storage: load from file
+        try
         {
-            messagesBranch = m_XRefTree->addPropTree("Messages",createPTree());
-            commit();
+            StringBuffer filepath(xrefPath);
+            addPathSepChar(filepath).append(branchName).append(".xml");
+
+            Owned<IFile> file = createIFile(filepath.str());
+            if (file->exists())
+            {
+                Owned<IFileIO> fileIO = file->open(IFOread);
+                if (fileIO)
+                {
+                    offset_t fileSize = file->size();
+                    if (fileSize > 0 && fileSize < XREF_MAX_FILE_SIZE)
```

Similar to previous comment, if file is greater than XREF_MAX_FILE_SIZE, it will be silently ignored.

---

will issue operator warning